pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,1274 @@
+}
+
+func (gqm *GroupQuotaManager) LockSetClusterTotalResource(totalRes v1.ResourceList) {
+	gqm.Lock.Lock()
+	defer gqm.Lock.Unlock()
+
+	delta := quotav1.Subtract(totalRes, gqm.totalResourceWithSys)
+	klog.V(3).Infof("SetClusterTotalResource totalRes:%s", totalRes)
+	gqm.updateClusterTotalResourceNoLock(delta)
+}
+
+func (gqm *GroupQuotaManager) LockUpdateClusterTotalResource(deltaRes v1.ResourceList) {
+	gqm.Lock.Lock()
+	defer gqm.Lock.Unlock()
+
+	klog.V(3).Infof("UpdateClusterResource deltaRes:%s", deltaRes)
+	gqm.updateClusterTotalResourceNoLock(deltaRes)
+}
+
+func (gqm *GroupQuotaManager) updateClusterTotalResourceNoLock(deltaRes v1.ResourceList) {
+	gqm.totalResourceWithSys = quotav1.Add(gqm.totalResourceWithSys, deltaRes)
+
+	totalResNoSys := quotav1.Subtract(gqm.totalResourceWithSys, gqm.sysQuotaUsed)
+
+	diffRes := quotav1.Subtract(totalResNoSys, gqm.totalResourceNoSys)
+
+	if !quotav1.IsZero(diffRes) {
+		gqm.totalResourceNoSys = totalResNoSys.DeepCopy()
+		gqm.quotaTreeWrapperMap[SystemQuotaName].LockUpdateClusterTotalResource(diffRes)
+		klog.V(3).Infof("UpdateClusterResource finish totalResourceNoSys:%v", gqm.totalResourceNoSys)
+	}
+}
+
+func (gqm *GroupQuotaManager) LockGetClusterTotalResourceWithSys() v1.ResourceList {
+	gqm.Lock.Lock()
+	defer gqm.Lock.Unlock()
+
+	return gqm.totalResourceWithSys.DeepCopy()
+}
+
+func (gqm *GroupQuotaManager) LockUpdateGroupFullRequest(quotaName QuotaNameType, fullRequest v1.ResourceList) {
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	gqm.updateGroupFullRequestNoLock(quotaName, fullRequest)
+}
+
+func (gqm *GroupQuotaManager) updateGroupFullRequestNoLock(quotaName QuotaNameType, fullRequest v1.ResourceList) {
+	curToParInfos := gqm.getCurToAllParentGroupQuotaInfoNoLock(quotaName)
+	allQuotaInfoLen := len(curToParInfos)
+	if allQuotaInfoLen <= 0 {
+		return
+	}
+
+	defer gqm.scopedLock4QuotaInfo(curToParInfos)()
+
+	quotaInfo := curToParInfos[0]
+	deltaReq := quotav1.Subtract(fullRequest, quotaInfo.CalculateInfo.Request)
+	if quotav1.IsZero(deltaReq) {
+		return
+	}
+	gqm.updateGroupDeltaRequestTopoRecursiveNoLock(quotaInfo, deltaReq, curToParInfos)
+}
+
+func (gqm *GroupQuotaManager) LockUpdateGroupDeltaRequest(quotaName QuotaNameType, deltaReq v1.ResourceList) {
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	gqm.updateGroupDeltaRequestNoLock(quotaName, deltaReq)
+}
+
+func (gqm *GroupQuotaManager) updateGroupDeltaRequestNoLock(quotaName QuotaNameType, deltaReq v1.ResourceList) {
+	curToAllParInfos := gqm.getCurToAllParentGroupQuotaInfoNoLock(quotaName)
+	allQuotaInfoLen := len(curToAllParInfos)
+	if allQuotaInfoLen <= 0 {
+		return
+	}
+
+	defer gqm.scopedLock4QuotaInfo(curToAllParInfos)()
+
+	quotaInfo := curToAllParInfos[0]
+	gqm.updateGroupDeltaRequestTopoRecursiveNoLock(quotaInfo, deltaReq, curToAllParInfos)
+}
+
+func (gqm *GroupQuotaManager) updateGroupDeltaRequestTopoRecursiveNoLock(quotaInfo *QuotaInfo, deltaReq v1.ResourceList,
```

Comment:
When this function updates the quota of a node, it also updates all parent nodes. The logic is quite clear, but the lock operation of treeWrapperPtr and directParTreeWrapperPtr is difficult to understand. Add some comments?

Reply:
get, will add more comments in functions like this.